app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +25,42 @@ def show
     }
   end
 
+  def raw
+    @version ||= version_collection.find(params[:id])
+
+    expires_in 1.year, public: true
+
+    if Archiver.external_archive_url?(@version.uri)
+      redirect_to @version.uri, status: 301 && return
+    elsif Archiver.store.contains_url?(@version.uri)
+      # Get the file
+      filename = File.basename(@version.uri)
+      upstream = Archiver.store.get_file(filename)
+      # Try to get the filetype, fall back on binary.
+      type = version_media_type(@version) || upstream.content_type || 'application/octet-stream'
```

Comment:
`upstream.content_type` is no longer a thing, right?

Reply:
If I recall correctly, it is in the s3 file objects. I'll verify that when I go check the s3 functionality. I named the (now removed) singleton for local files to align with method of s3 objects. I can drop this anyway if you want, since it sounds like the plan is to backfill versions with info.

Reply:
`get_file` should return a string, not annotated with anything special: https://github.com/edgi-govdata-archiving/web-monitoring-db/blob/459c23211455966826b1470562e1ebe81294d431/lib/file_storage/s3.rb#L25-L27

(Maybe it would be a good improvement to return some more special kind of object, but that should probably be out of scope here.)

Reply:
Or: none of the methods in `FileStorage::S3` should give you back any S3 types. It should have the exact same public API as `FileStorage::LocalFile`.

Reply:
Interesting. I may have be recalling some bizarre behavior that I myself introduced on a previous iteration of this. I'll just kill `content_type` as requested.